src/training.py

```diff
@@ -133,11 +134,13 @@ def train_model(
     lr = trainer_params['lr']
     crit_type = trainer_params['criterion']
 
-    model = model_class(
-        d_features=x_dim,
-        device=device,
-        **model_params
-    )
+    if model is None:
+        model = model_class(
+            d_features=x_dim,
```

Comment:
has the abstract class been updated to expect this?

Reply:
Either every model needs the d_features or this needs to be forced in the abstract model, otherwise we will get failures for the versions of the models that do not have the d_features parameter.